onnxruntime/core/providers/rocm/math/einsum_utils/einsum_auxiliary_ops.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/shared_library/provider_api.h"
+
+namespace onnxruntime {
+namespace concurrency {
+class ThreadPool;
+}
+}  // namespace onnxruntime
+
+#include "einsum_auxiliary_ops.h"
+
+namespace onnxruntime {
+
+namespace EinsumOp {
+
+namespace DeviceHelpers {
+
+namespace CudaDeviceHelpers {
+
+// ROCM EP specific Data copy helper
+Status DataCopy(const Tensor& input, Tensor& output, void* einsum_rocm_assets) {
+  ORT_ENFORCE(output.SizeInBytes() == input.SizeInBytes(),
+              "Einsum op: The candidate output does not match the actual output's shape");
+  // There are no string tensors in Einsum's case - so safely use memcpy
+  // TODO: Currently, triggers copy on stream 0, investigate if we can still do that
```

Comment:
I guess this TODO is no longer relevant. We are launching the copy on the same stream as the kernel stream. Could you please remove this TODO and the corresponding one in the CUDA implementation ?